4.assess_fibrosis_reversal/scripts/0.apply_models.py

```diff
@@ -0,0 +1,259 @@
+label_encoder_url = "https://github.com/WayScience/cellpainting_predicts_cardiac_fibrosis/raw/refs/heads/main/5.machine_learning/encoder_results/label_encoder_log_reg_fs_plate_4.joblib"
+
+# Create the models directory if it doesn't exist
+models_dir = pathlib.Path("models")
+models_dir.mkdir(exist_ok=True)
+
+# Create the models directory if it doesn't exist
+label_encoder_dir = pathlib.Path("label_encoder")
+label_encoder_dir.mkdir(exist_ok=True)
+
+# Download the final and shuffled models
+final_model_path = download_file(final_model_url, models_dir)
+shuffled_model_path = download_file(shuffled_model_url, models_dir)
+encoder_path = download_file(label_encoder_url, label_encoder_dir)
+
+
+# ## Get list of features for models to filter data
+
+# In[4]:
+
+
+# Define the GitHub raw link for the feature selected parquet file for the plate used with the models
+github_url = "https://github.com/WayScience/cellpainting_predicts_cardiac_fibrosis/raw/refs/heads/main/3.process_cfret_features/data/single_cell_profiles/localhost231120090001_sc_feature_selected.parquet"
+
+# Load the parquet file into memory (not locally)
+response = requests.get(github_url)
+response.raise_for_status()  # Ensure the request was successful
+
+# Convert the response content into a BytesIO object for pandas to read
+parquet_file = BytesIO(response.content)
+
+# Load the Parquet file into a pandas dataframe
+df = pd.read_parquet(parquet_file)
+
+# Get the column names that do not start with "Metadata"
+model_features = [col for col in df.columns if not col.startswith("Metadata")]
+
+# Output the number of columns to check if it matches correctly
+print(len(model_features))
+
+
+# ## Set paths and variables
+
+# In[5]:
+
+
+# Directory for probability data to be saved
+prob_dir = pathlib.Path("./prob_data")
+prob_dir.mkdir(exist_ok=True)
+
+# Directory with normalized plate datasets  
+data_dir = pathlib.Path("../3.preprocessing_features/data/single_cell_profiles")
+
+# Use rglob to search for files with the suffix *_sc_normalized.parquet
+parquet_files = list(data_dir.rglob("*_sc_normalized.parquet"))
+
+# Load all matching parquet files into a list of DataFrames
+dfs = [pd.read_parquet(file) for file in parquet_files]
+
+# Print the file paths and check the content if needed
+for file in parquet_files:
+    print(f"Loaded file: {file}")
+
+
+# ## Filter the data for the model features and add to dictionary to process
+
+# In[6]:
+
+
+# Dictionary to store plate data
+plate_data_dict = {}
+
+# Loop through and process each parquet file
+for parquet_file in parquet_files:
+    # Extract the plate name from the file name
+    plate_name = parquet_file.name.split('_')[0]
+    print(f"Processing plate: {plate_name}")
+
+    # Load the Parquet file
+    df = pd.read_parquet(parquet_file)
+
+    # Drop rows with NaN values in feature columns that the model uses
+    df = df.dropna(subset=model_features)
```

Comment:
We would still need the metadata loaded in here since we want to include the metadata for each single-cell predicted probability in the end. So in this context, it would be better off to load in everything at once and then specify after.